pkg/core/stateroot/module.go

```diff
@@ -181,6 +187,57 @@ func (s *Module) JumpToState(sr *state.MPTRoot) error {
 	return nil
 }
 
+const gcBatchSize = 100000
+
+// GC performs garbage collection.
+func (s *Module) GC(index uint32, store storage.Store) time.Duration {
+	if !s.mode.GC() {
+		panic("stateroot: GC invoked, but not enabled")
+	}
+	var oldKey = []byte{byte(storage.DataMPT)}
+	var finished bool
+	var removed int
+	s.log.Info("starting MPT garbage collection", zap.Uint32("index", index))
+	start := time.Now()
+	for !finished {
+		var batched int
+		b := store.Batch()
+		finished = true // Speculatively.
+		store.Seek(storage.SeekRange{
+			Prefix: oldKey[:1],
+			Start:  oldKey[1:],
+		}, func(k, v []byte) bool {
+			if !mpt.IsActiveValue(v) {
+				h := binary.LittleEndian.Uint32(v[len(v)-4:])
+				if h > index {
+					return true
+				}
+				b.Delete(k)
```

Comment:
Boltdb supports deleting items from cursor in place. We don't have a proper abstraction in Store RN, but I wonder how it could affect performance (no allocations for batch and ~twice less key-accesses).

Reply:
Leveldb has no such support in iterator, though it stiil can turn out positive because of limited cache size.

Reply:
At the moment we're using `View` even for Bolt which is RO, but yeah, a separate GC interface might be interesting (#2096 will also iterate over keys).